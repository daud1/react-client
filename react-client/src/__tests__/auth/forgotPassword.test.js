import React from 'react';
import { shallow, render, mount } from 'enzyme';
import shallowToJson from 'enzyme-to-json';
import moxios from 'moxios';

import ForgotPassword from '../../components/auth/forgotPassword.component';

describe ('ForgottenPassword Component', () => {
    it('should render without crashing', () => {
        const wrapper = shallow(<ForgotPassword />);
        expect(wrapper.exists(<div></div>)).toBe(true)
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<ForgotPassword />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('handles successful axios calls properly', (done) => {
        const wrapper = shallow(<ForgotPassword />);

        wrapper.find('TextField[name="email"]').simulate('change', { target: { value: 'user@email.com' } });
        wrapper.find('form').simulate('submit', { preventDefault() {} });

        moxios.stubRequest('http://localhost:5000/auth/forgot-password', {
            status: 200,
            'MSG': 'Password reset link sent!'
        });
        moxios.wait(() => {
            expect(wrapper.html()).toContain('Password reset link sent!');
            done();
        });
    });
})