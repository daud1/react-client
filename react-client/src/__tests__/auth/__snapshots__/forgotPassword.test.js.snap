// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgottenPassword Component matches snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <p>
       Enter your email address below to recieve a password reset link.
    </p>
    <input
      name="email"
      placeholder="Enter your email address"
      type="email"
    />
    <button
      type="submit"
    >
      Send Reset Link
    </button>
  </form>
</div>
`;
